Sources/AppcuesKit/Presentation/Extensions/View+Appcues.swift

```diff
@@ -129,13 +129,21 @@ extension View {
 
 @available(iOS 13.0, *)
 extension Text {
-    func applyTextStyle(_ style: AppcuesStyle) -> some View {
+    func applyTextStyle(_ style: AppcuesStyle, text: String) -> some View {
         self
             .ifLet(style.letterSpacing) { view, val in
                 view.kerning(val)
             }
             .ifLet(style.textAlignment) { view, val in
                 view.multilineTextAlignment(val)
             }
+            // The following two lines allow text to scale down in cases where it is too
+            // large to fit in the bounds available, and otherwise unnatural clipping
+            // or wrapping would occur. For example, a large emoji in a side-by-side layout.
+            // SwiftUI requires a line limit to be set for the minimumScaleFactor to be applied,
+            // and it must be set to 1 line in the case of a single large character for scaling
+            // to be applied.
+            .lineLimit(text.count == 1 ? 1 : Int.max)
+            .minimumScaleFactor(text.count == 1 ? 0.1 : 0.5)
```

Comment:
Guesswork here - seems like maybe shrinking a font to half its intended size would be the minimum desired in most cases.  In the case of a single character, or emoji, shrinking until it fits (or the extreme 10% size case) may be desired.

Reply:
@mmaatttt something like 
```
.minimumScaleFactor(min(10.0 / (model.style?.fontSize ?? 17.0), 1.0))
```
that would let things scale down to 10pt font, regardless of original size?

Reply:
Thanks, was trying to find the right system prop to reference for that. updated!